library_builders.sh

```diff
@@ -200,11 +239,29 @@ function build_hdf5 {
     file hdf5*.tar.gz
     tar -xzf hdf5*.tar.gz
     rm hdf5*.tar.gz
-    cd hdf5-*
-    ./configure --disable-parallel --disable-shared --enable-static --prefix ${BUILD_PREFIX}
-    make -j${CPU_COUNT}
-    make install
-    cd -
+
+    CMAKE_CROSSCOMPILING_EMULATOR=""
+    if [[ "${CMAKE_OSX_ARCHITECTURES-}" == "arm64" ]]; then
+        CMAKE_CROSSCOMPILING_EMULATOR="qemu-system-aarch64;-M;virt"
+    fi
+
+    PY_BIN=$(which python3)
+    CMAKE_BIN="$(${PY_BIN} -m pip show cmake 2>/dev/null | grep Location | cut -d' ' -f2)/cmake/data/bin/"
+    PATH=${CMAKE_BIN}:${PATH} cmake          \
+      -S hdf5-*                              \
+      -B build-hdf5                          \
+      -DBUILD_TESTING=OFF                    \
+      -DBUILD_SHARED_LIBS=OFF                \
+      -DBUILD_STATIC_LIBS=ON                 \
+      -DHDF5_BUILD_EXAMPLES=OFF              \
+      -DHDF5_BUILD_FORTRAN=OFF               \
+      -DHDF5_BUILD_TOOLS=OFF                 \
+      -DHDF5_BUILD_UTILS=OFF                 \
+      -DHDF5_INSTALL_CMAKE_DIR=share/cmake/hdf5 \
+      -DCMAKE_CROSSCOMPILING_EMULATOR="${CMAKE_CROSSCOMPILING_EMULATOR}" \
+      -DCMAKE_INSTALL_PREFIX=${BUILD_PREFIX}
+    cmake --build build-hdf5 -j ${CPU_COUNT}
+    cmake --build build-hdf5 --target install
```

Comment:
Has issues because they do a lot of `try_run` during config and compilation.
```suggestion
    cd hdf5-*

    # macOS cross-compile
    #   https://github.com/conda-forge/hdf5-feedstock/blob/cbbd57d58f7f5350ca679eaad49354c11dd32b95/recipe/build.sh#L53-L80
    if [[ "${CMAKE_OSX_ARCHITECTURES-}" == "arm64" ]]; then
        export ac_cv_sizeof_long_double=8
        export hdf5_cv_ldouble_to_long_special=no
        export hdf5_cv_long_to_ldouble_special=no
        export hdf5_cv_ldouble_to_llong_accurate=yes
        export hdf5_cv_llong_to_ldouble_correct=yes
        export hdf5_cv_disable_some_ldouble_conv=no
        export hdf5_cv_system_scope_threads=yes
        export hdf5_cv_printf_ll="l"
        export PAC_FC_MAX_REAL_PRECISION=15
        export PAC_C_MAX_REAL_PRECISION=17
        export PAC_FC_ALL_INTEGER_KINDS="{1,2,4,8,16}"
        export PAC_FC_ALL_REAL_KINDS="{4,8}"
        export H5CONFIG_F_NUM_RKIND="INTEGER, PARAMETER :: num_rkinds = 2"
        export H5CONFIG_F_NUM_IKIND="INTEGER, PARAMETER :: num_ikinds = 5"
        export H5CONFIG_F_RKIND="INTEGER, DIMENSION(1:num_rkinds) :: rkind = (/4,8/)"
        export H5CONFIG_F_IKIND="INTEGER, DIMENSION(1:num_ikinds) :: ikind = (/1,2,4,8,16/)"
        export PAC_FORTRAN_NATIVE_INTEGER_SIZEOF="                    4"
        export PAC_FORTRAN_NATIVE_INTEGER_KIND="           4"
        export PAC_FORTRAN_NATIVE_REAL_SIZEOF="                    4"
        export PAC_FORTRAN_NATIVE_REAL_KIND="           4"
        export PAC_FORTRAN_NATIVE_DOUBLE_SIZEOF="                    8"
        export PAC_FORTRAN_NATIVE_DOUBLE_KIND="           8"
        export PAC_FORTRAN_NUM_INTEGER_KINDS="5"
        export PAC_FC_ALL_REAL_KINDS_SIZEOF="{4,8}"
        export PAC_FC_ALL_INTEGER_KINDS_SIZEOF="{1,2,4,8,16}"
        export hdf5_disable_tests="--enable-tests=no"
    fi

     ./configure --disable-parallel --disable-shared --enable-static --prefix ${BUILD_PREFIX}
    make -j${CPU_COUNT}
    make install
    cd -
```

Reply:
or patch as in https://github.com/h5py/h5py/pull/2099